src/coreclr/jit/emit.h

```diff
@@ -2989,9 +2991,9 @@ inline emitAttr emitActualTypeSize(T type)
 /* static */ inline emitter::opSize emitter::emitEncodeSize(emitAttr size)
 {
     assert(size == EA_1BYTE || size == EA_2BYTE || size == EA_4BYTE || size == EA_8BYTE || size == EA_16BYTE ||
-           size == EA_32BYTE);
+           size == EA_32BYTE || size == EA_64BYTE);
 
-    return emitSizeEncode[((int)size) - 1];
+    return emitSizeEncode[genLog2(size)];
```

Comment:
Why the switch to `genLog2`? Would that impact throughput?

Reply:
It was a suggested change by me to keep the `emitSizeEncode` table smaller since it would grow to 64-entries and be 90% `OPSIZE_INVALID`.

It's possible that it's negatively impacting throughput. That being said, `genLog2` is just emitting `BitScanForward` which is a 3 cycle instruction, so I wouldn't expect it to have terrible impact. I would have expected `emitSizeEncode` taking up 4 cache lines would have been worse.

Notably we don't actually "need" a memory lookup here. `genLog2(size) - 1` by itself should be sufficient since `emmitter:opSize` starts at `OPSZ1 = 0` and goes to `OPSZ64 = 6`

Reply:
Removed the lookup